include/fiction/algorithms/physical_design/planar_layout_from_network_embedding.hpp

```diff
@@ -0,0 +1,1184 @@
+ * @return     Orientation code (0, 1, or 3) defining the level's initial direction.
+ */
+template <typename Ntk>
+int calculate_start_orientation(const Ntk& ntk, const uint32_t lvl)
+{
+    int orientation = 0;
+    if (lvl == 0)
+    {
+        return orientation;
+    }
+    ntk.foreach_node_in_rank(lvl,
+                             [&ntk, &orientation](const auto& n)
+                             {
+                                 if (orientation > 0)
+                                 {
+                                     return;
+                                 }
+                                 if (ntk.fanin_size(n) == 2)
+                                 {
+                                     orientation = 4;
+                                     return;
+                                 }
+                                 if (const auto fc = fanins(ntk, n); fc.fanin_nodes.size() == 1)
+                                 {
+                                     const auto& pre = fc.fanin_nodes[0];
+                                     if (ntk.is_fanout(pre) && ntk.fanout_size(pre) == 2)
+                                     {
+                                         orientation = 1;
+                                         return;
+                                     }
+                                 }
+                                 if (ntk.fanout_size(n) == 2)
+                                 {
+                                     orientation = 1;
+                                     return;
+                                 }
+                             });
+    return (orientation == 0) ? 0 : (orientation - 1);
+    // instead of just returning 0, here the buffers could be wired in a way they are closer together and hence overhead
+    // produced by two input nodes is reduced
+}
+
+/**
+ * Computes orientation and routing line variables for a given network level.
+ *
+ * For each node in the specified level, the function determines:
+ *  - The node's orientation, based on its fan-in structure, gap spacing, and connection type.
+ *  - Whether new routing lines must be introduced, derived from lookup tables for buffer and fan-out configurations.
+ *
+ * @tparam Ntk  Logic network type.
+ * @tparam Lyt  Layout type.
+ * @param ntk       Logic network to analyze.
+ * @param node2pos  Mapping from network nodes to layout tile positions.
+ * @param lvl       Level index for which to compute the variables.
+ * @return          Tuple containing:
+ *                   - `orientation`: orientation values for each node in the level.
+ *                   - `new_lines`: flags indicating newly added routing lines.
+ */
+template <typename Ntk, typename Lyt>
+std::tuple<std::vector<uint64_t>, std::vector<uint64_t>>
+compute_pr_variables(const Ntk& ntk, const mockturtle::node_map<mockturtle::signal<Lyt>, Ntk>& node2pos,
+                     const uint32_t& lvl)
```

Comment:
```suggestion
                     const uint32_t lvl)
```
Don't pass trivial types by reference unless you intend to manipulate them (non-`const`).